CAIPs/caip-211.md

```diff
@@ -0,0 +1,275 @@
+        "rpcEndpoints": ["https://node1.42069-chain.org/"]
+      },
+      "cosmos": {
+        ...
+      }
+    },
+    "optionalScopes":{
+      "eip155:42161": {
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "get_balance", "personal_sign"],
+        "events": ["accountsChanged", "chainChanged"]
+    },
+    "sessionProperties": {
+      "expiry": "2022-12-24T17:07:31+00:00",
+      "caip154-mandatory": "true"
+    }         
+  }
+}
+```
+
+### Response
+
+The wallet can respond to this method with either a success result or an error message.
+
+#### Success
+
+A succesful result will be a conformant successful result according to
+[CAIP-25][], with the additional presence (if authorized) of the new properties
+in one or more scope objects.
+
+An example of a successful response follows:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "sessionId": "0xdeadbeef",
+    "sessionScopes": {
+      "eip155": {
+        "chains": ["eip155:1", "eip155:137"],
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "get_balance", "eth_sign", "personal_sign"]
+        "events": ["accountsChanged", "chainChanged"],
+        "accounts": ["eip155:1:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb", "eip155:137:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"]
+      },
+      "eip155:10": {
+        "methods": ["get_balance"],
+        "events": ["accountsChanged", "chainChanged"],
+        "accounts:" []
+      },
+      "eip155:42069": {
+        "methods": ["get_balance", "chainChanged", "42069_sEcReTbAlAnCe"],
+        "rpcDocuments": ["https://raw.githubusercontent.com/ethereum/execution-apis/assembled-spec/refs-openrpc.json", "https://openrpc.42069-chain.org/"],
+        "rpcEndpoints": ["https://node1.42069-chain.org/"]
+      }
+        //...
+    },      
+    "sessionProperties": {
+      "expiry": "2022-11-31T17:07:31+00:00"          
+    }
+  }
+}
+```
+
+Note:
+* The response re-ordered the RPC documents array, which still allows the
+  42069-chain docs to define new methods such as the namespaced secret-balance
+  method example. However, since the ordinality of the array prioritizes earlier
+  authorities in any case where two define the same term, the chain-specific
+  authority is effectively prevented from redefining any methods defined in the
+  eip155-wide standard, even in the limited scope of the 42069 chain. This is
+  purely for illustrative purposes and not normative.
+
+#### Failure States
+
+##### TODO: 
+
+An example of an error response should match the following format:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "error": {
+    "code": 5000,
+    "message": "Unknown error"
+  }
+}
+```
+
+The additional error messages codes to be supported in addition to CAIP-25s are the following:
+
+##### TODO: replace following with novel errors
+
+Possible error messages (to be discussed with WG):
+- rpcDocuments not conformant syntactically (not openRPC, not served as mime type JSON, etc)
```

Comment:
There's an OpenRPC doc to describe the OpenRPC spec, that would be helpful 😄

Reply:
Sure! The error message could even LINK to it (some WC error messages currently link to CAIP URLs!) although I'm not sure where to draw the normativity line on things like error messages